packages/smooth_app/lib/pages/preferences_v2/roots/preferences_root.dart

```diff
@@ -0,0 +1,188 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:sliver_tools/sliver_tools.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/l10n/app_localizations.dart';
+import 'package:smooth_app/pages/preferences_v2/cards/preference_card.dart';
+import 'package:smooth_app/pages/preferences_v2/tiles/external_search_tiles/external_search_preference_tile.dart';
+import 'package:smooth_app/pages/preferences_v2/tiles/navigation_preference_tile.dart';
+import 'package:smooth_app/pages/preferences_v2/tiles/preference_tile.dart';
+import 'package:smooth_app/pages/product/common/search_helper.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+import 'package:smooth_app/widgets/v2/smooth_leading_button.dart';
+import 'package:smooth_app/widgets/v2/smooth_topbar2.dart';
+
+class PreferencesRootSearchController extends SearchHelper {
+  PreferencesRootSearchController();
+
+  String? query;
+
+  @override
+  void search(
+    BuildContext context,
+    String? keywords, {
+    required Function(String) searchQueryCallback,
+  }) {
+    query = keywords;
+    notifyListeners();
+  }
+
+  @override
+  String getHintText(AppLocalizations appLocalizations) {
+    return appLocalizations.preferences_app_bar_search_hint;
+  }
+
+  @override
+  String get historyKey => 'preferences_root_search_history';
+}
+
+/// Base class for all preference roots.
+/// [PreferencesRoot] corresponds to a page where each root can have its own
+/// app bar and cards.
+/// The [getCards] method should be overridden to provide the cards for the root.
+/// The [getExternalSearchTiles] method can be overridden to provide external
+/// search tiles for the root which will be displayed in the search results only.
+/// The [buildAppBar] method can be overridden to provide a custom app bar for
+/// the root.
+/// The [PreferencesRoot] also handles the search logic. When the user searches
+/// for a tile, the cards are replaced by individual tiles that match the search
+/// query.
+/// The search query is performed deep in the widget tree.
+abstract class PreferencesRoot extends StatelessWidget {
+  const PreferencesRoot({
```

Comment:
Dev preferences show be ignored if dev mode is disabled

Reply:
If you mean Dev mode tiles should not show in search if dev mode is not active, it is already taken care of as the searchTiles method crawls the instantiated tree which does not contain DevModeRoot if inactive.